### PR TITLE
Item Editor on Double Click

### PR DESCRIPTION
**Pull request by gkahiu**
UX enhancement for showing the editor dialog on double clicking an NCS pathway, implementation model or priority layer item in the list widget.

### CODE REVIEW

**Review by Samweli — APPROVED** [review]

**Comment by Samweli** [comment]
@gkahiu this looks good, can you add the missing docstrings for methods parameters then it will good for merge, thanks.

**Comment by gkahiu** [comment]
Thanks @Samweli. Which ones exactly as most are internal methods not in the public API?

File: src/cplus_plugin/gui/model_component_widget.py [hunk 1]
```diff
@@ -209,6 +211,31 @@ def _on_selection_changed(
         """
         self._update_ui_on_selection_changed()
 
+    def _on_double_click_item(self, index: QtCore.QModelIndex):
```
Comment (Samweli) [inline src/cplus_plugin/gui/model_component_widget.py:214]: @gkahiu this one.

File: src/cplus_plugin/gui/model_component_widget.py [hunk 2]
```diff
@@ -371,11 +398,17 @@ def _on_edit_item(self):
             return
 
         item = selected_items[0]
+        self._edit_ncs_pathway_item(item)
 
+    def _handle_double_click(self, item: NcsPathwayItem):
```
Comment (Samweli) [inline src/cplus_plugin/gui/model_component_widget.py:403]: same as above

File: src/cplus_plugin/gui/model_component_widget.py [hunk 3]
```diff
@@ -371,11 +398,17 @@ def _on_edit_item(self):
...
+        """Show editor dialog."""
+        self._edit_ncs_pathway_item(item)
+
+    def _edit_ncs_pathway_item(self, item: NcsPathwayItem):
+        """Shows dialog for editing an item."""
```
Comment (Samweli) [inline src/cplus_plugin/gui/model_component_widget.py:408]: same as above

**Comment by gkahiu** [comment]
Hi @Samweli, I've pushed the changes as requested.

Also to clarify, when do we need docstrings - public, private and/or protected methods/attributes? Currently, I have just been doing it for public ones.

File: src/cplus_plugin/gui/model_component_widget.py (hunk 1 above)
Comment (gkahiu) [reply src/cplus_plugin/gui/model_component_widget.py:214]: Done.

File: src/cplus_plugin/gui/model_component_widget.py (hunk 2 above)
Comment (gkahiu) [reply src/cplus_plugin/gui/model_component_widget.py:407]: Done.

File: src/cplus_plugin/gui/model_component_widget.py (hunk 3 above)
Comment (gkahiu) [reply src/cplus_plugin/gui/model_component_widget.py:416]: Done.

**Review by Samweli — APPROVED** [review]

**Pull request merged**